backend/services/messaging/message_service.py

```diff
@@ -706,8 +706,17 @@ def get_all_messages(
             query = query.filter(Message.read == (True if status == "read" else False))
 
         if message_type:
-            message_type_filters = map(int, message_type.split(","))
-            query = query.filter(Message.message_type.in_(message_type_filters))
+            message_type_filters = [
+                key.strip().lower() for key in message_type.split(",")
+            ]
+
+            filter_values = [
+                key.value
+                for key in MessageType
+                if key.name.lower() in message_type_filters
+            ]
+
+            query = query.filter(Message.message_type.in_(filter_values))
```

Comment:
Can you explain this change?

Reply:
While testing, I could not retrieve any notifications if I specified the message type. Not when I used numbers (for the mapped integer value assigned to a type under the MessageType enum class), e.g. 1 (system), 2 (broadcast)., etc. Nor when I typed in the string value itself, e.g. system/SYSTEM/System. That was with the previous implementation.

So, this new implementation lets a user retrieve the specific messages when they filter by the string value of the message type, e.g. system, mention_notification, broadcast, etc. 

Could you test to confirm if my findings were accurate or not? @Aadesh-Baral

Reply:
This change has introduced a problem on the frontend. In our frontend code, we have four notification types that are associated with different combinations of MessageTypes. You can find more information about this issue in the following link: [GitHub Issue Comment](https://github.com/hotosm/tasking-manager/pull/5676#issuecomment-1506579190).

For instance, when applying a teams filter in the frontend, a query parameter is passed as "types=6,7,11" in the NotificationsAllAPI. To filter and retrieve the notifications for the provided message types on the backend, this value needs to be split by comma (',') and converted to integers.

However, due to this change, the value is being split correctly but is not being converted to integers, resulting in empty results being returned.

Reply:
Noted. I have adjusted the tests, removed this change, and added tests for the sortBy and sortDirection query parameters. 
Sidenote: For future reference, we should probably include in the documentation the expected values for the message_type filter and what type the respective numerical values correspond to.

Reply:
> For future reference, we should probably include in the documentation the expected values for the message_type filter and what type the respective numerical values correspond to.

Sounds good

Reply:
> For future reference, we should probably include in the documentation the expected values for the message_type filter and what type the respective numerical values correspond to.

@d-rita Let's do this in this PR itself.